src/ReflectionProperty.php

```diff
@@ -60,13 +68,31 @@ public function __construct(
         Property $propertyType = null,
         PropertyProperty $propertyNode = null
     ) {
-        $this->className    = $className;
+        $this->className        = $className;
+        $this->propertyName     = $propertyName;
         if (!$propertyType || !$propertyNode) {
-            list ($propertyType, $propertyNode) = ReflectionEngine::parseClassProperty($className, $propertyName);
+            $oneNodeProvided = $propertyType || $propertyNode;
```

Comment:
No, in this case I was specifically referring to the quantity 1.  This is to catch an invalid case, as neither or both of these parameters should be non-null. This would probably be more readable if I changed it to an `xor` operator.

Reply:
Can't agree with you. Because when both parameter are present, then this value will be `true`. So it should be named `atLeastOneNode`. For exactly one node it should be calculated as `($propertyType && !$propertyNode) || (!$propertyType && $propertyNode)`

So, still need to fix naming or expression to match.

Reply:
It's possible that my logic is broken, but we should only be in this block if one or both values are null (or falsey), so they can't both be non-null. (This may be confusing because this happens before the assignment which is in red above it.) I will change to `xor` for clarity.